tests/transport/test_send_recv.cpp

```diff
@@ -0,0 +1,355 @@
+#include <iostream>
+#include <iomanip>
+#include <ghex/threads/none/primitives.hpp>
+#include <ghex/transport_layer/message_buffer.hpp>
+#include <ghex/transport_layer/shared_message_buffer.hpp>
+#include <ghex/common/timer.hpp>
+#include <gtest/gtest.h>
+
+#ifdef GHEX_TEST_USE_UCX
+#include <ghex/transport_layer/ucx/context.hpp>
+using transport = gridtools::ghex::tl::ucx_tag;
+#else
+#include <ghex/transport_layer/mpi/context.hpp>
+using transport = gridtools::ghex::tl::mpi_tag;
+#endif
+
+using threading = gridtools::ghex::threads::none::primitives;
+using context_type = gridtools::ghex::tl::context<transport, threading>;
+using communicator_type = typename context_type::communicator_type;
+
+#define NITERS 100000
+
+
+template<typename MsgType, typename CommType>
+auto test_ring_send_recv_ft(CommType& comm, MsgType &&rmsg, MsgType &&smsg) 
+{
+    gridtools::ghex::timer timer;
+    int *data_ptr;
+    int rank = comm.rank();
+    int size = comm.size();
+    int speer_rank = (rank+1)%size;
+    int rpeer_rank = (rank-1)%size;
+    if(rpeer_rank<0) rpeer_rank = size-1;
+
+    data_ptr = reinterpret_cast<int*>(smsg.data());
+    *data_ptr = rank;
+
+    timer.tic();
+    communicator_type::future<void> sreq, rreq;
+    for(int i=0; i<NITERS; i++){
+
+        rreq = comm.recv(rmsg, rpeer_rank, 1);
+        sreq = comm.send(smsg, speer_rank, 1);
+        while(!(rreq.ready() && sreq.ready()));
+
+        data_ptr = reinterpret_cast<int*>(rmsg.data());
+        EXPECT_TRUE(*data_ptr == rpeer_rank);
+    }
+    const auto t = timer.stoc();
+    if(rank==0)
+    {
+        std::cout << "time:       " << t/1000000 << "s\n";
+    }
+}
+
+template<typename MsgType, typename CommType>
+auto test_ring_send_recv_ft(CommType& comm) 
+{
+    MsgType smsg(4), rmsg(4);
+    test_ring_send_recv_ft(comm, std::move(rmsg), std::move(smsg));
```

Comment:
no move here, future-based calls always take the message by reference and you as caller have to make sure the message stays alive
```suggestion
    test_ring_send_recv_ft(comm, rmsg, smsg);
```